Remove the support for JPEG2000

shallawa opened the pull request:
#### 78cb9d96aa82cb8cfb1ecb6a1275efabfab31fab
<pre>
Remove the support for JPEG2000
<a href="https://bugs.webkit.org/show_bug.cgi?id=178758">https://bugs.webkit.org/show_bug.cgi?id=178758</a>
<a href="https://rdar.apple.com/35161822">rdar://35161822</a>

Reviewed by Michael Catanzaro.

Right now, no other browser is supporting JPEG2000. So there&apos;s no need for WebKit
to keep supporting it.

Clients of WebView and WKWebView can still force supporting JPEG2000 through the
private configuration/preference &apos;additionalSupportedImageTypes&apos;.

* LayoutTests/fast/images/image-formats-support-expected.txt:
* LayoutTests/fast/images/image-formats-support.html:
* LayoutTests/fast/images/resources/100x100-red-jp2-renamed.png: Added.
* LayoutTests/fast/images/resources/100x100-red.jp2: Added.
* Source/WebCore/platform/ImageDecoders.cmake:
* Source/WebCore/platform/MIMETypeRegistry.cpp:
* Source/WebCore/platform/graphics/cg/UTIRegistry.cpp:
(WebCore::defaultSupportedImageTypes):
* Source/WebCore/platform/image-decoders/ScalableImageDecoder.cpp:
(WebCore::ScalableImageDecoder::create):
* Source/WebCore/platform/image-decoders/jpeg2000/JPEG2000ImageDecoder.cpp: Removed.
* Source/WebCore/platform/image-decoders/jpeg2000/JPEG2000ImageDecoder.h: Removed.
* Source/cmake/OptionsGTK.cmake:
* Source/cmake/OptionsWPE.cmake:
* Tools/TestWebKitAPI/TestWebKitAPI.xcodeproj/project.pbxproj:
* Tools/TestWebKitAPI/Tests/WebKitCocoa/100x100-red.jp2: Added.
* Tools/TestWebKitAPI/Tests/WebKitCocoa/AdditionalSupportedImageTypes.mm:
(TEST):
* Tools/TestWebKitAPI/Tests/mac/AdditionalSupportedImageTypes.html:
* Tools/TestWebKitAPI/Tests/mac/AdditionalSupportedImageTypes.mm:
(TestWebKitAPI::runTest):
(TestWebKitAPI::TEST):
* Tools/gtk/jhbuild.modules:
* Tools/wpe/jhbuild.modules:

Canonical link: <a href="https://commits.webkit.org/273978@main">https://commits.webkit.org/273978@main</a>
</pre>
<!--EWS-Status-Bubble-Start-->
https://github.com/WebKit/WebKit/commit/95cdb4dc849acc5877e0f208b0116983c02a8ba7

| Misc | iOS, tvOS & watchOS  | macOS  | Linux |  Windows |
| ----- | ---------------------- | ------- |  ----- |  --------- |
| [✅ 🧪 style](https://ews-build.webkit.org/#/builders/38/builds/37337 "Passed style check") | [✅ 🛠 ios](https://ews-build.webkit.org/#/builders/48/builds/16222 "Built successfully") | [✅ 🛠 mac](https://ews-build.webkit.org/#/builders/14/builds/39609 "Built successfully") | [✅ 🛠 wpe](https://ews-build.webkit.org/#/builders/5/builds/39872 "Built successfully") | [✅ 🛠 wincairo](https://ews-build.webkit.org/#/builders/32/builds/33294 "Built successfully") 
| [✅ 🧪 bindings](https://ews-build.webkit.org/#/builders/9/builds/38645 "Passed tests") | [✅ 🛠 ios-sim](https://ews-build.webkit.org/#/builders/49/builds/18818 "Built successfully") | [✅ 🛠 mac-AS-debug](https://ews-build.webkit.org/#/builders/51/builds/13372 "Built successfully") | [✅ 🧪 wpe-wk2](https://ews-build.webkit.org/#/builders/34/builds/31733 "Passed tests") | 
| [✅ 🧪 webkitperl](https://ews-build.webkit.org/#/builders/11/builds/37902 "Passed tests") | [✅ 🧪 ios-wk2](https://ews-build.webkit.org/#/builders/47/builds/13666 "Passed tests") | [✅ 🧪 api-mac](https://ews-build.webkit.org/#/builders/18/builds/32802 "Passed tests") | [✅ 🧪 api-wpe](https://ews-build.webkit.org/#/builders/41/builds/11897 "Passed tests") | 
| | [✅ 🧪 ios-wk2-wpt](https://ews-build.webkit.org/#/builders/42/builds/11886 "Passed tests") | [✅ 🧪 mac-wk1](https://ews-build.webkit.org/#/builders/10/builds/33445 "Passed tests") | [✅ 🛠 gtk](https://ews-build.webkit.org/#/builders/2/builds/41135 "Built successfully") | 
| [✅ 🛠 🧪 jsc](https://ews-build.webkit.org/#/builders/20/builds/31180 "Built successfully and passed tests") | [✅ 🧪 api-ios](https://ews-build.webkit.org/#/builders/13/builds/33772 "Passed tests") | [✅ 🧪 mac-wk2](https://ews-build.webkit.org/#/builders/36/builds/33879 "Passed tests") | [✅ 🧪 gtk-wk2](https://ews-build.webkit.org/#/builders/1/builds/37793 "Passed tests") | 
| [✅ 🛠 🧪 jsc-arm64](https://ews-build.webkit.org/#/builders/12/builds/37027 "Built successfully and passed tests") | [  ~~🛠 tv~~](https://ews-build.webkit.org/#/builders/44/builds/12455 "The change is no longer eligible for processing. Pull Request was already closed when EWS attempted to process it.") | [✅ 🧪 mac-AS-debug-wk2](https://ews-build.webkit.org/#/builders/50/builds/10024 "Passed tests") | [✅ 🧪 api-gtk](https://ews-build.webkit.org/#/builders/21/builds/35940 "Passed tests") | 
| | [  ~~🛠 tv-sim~~](https://ews-build.webkit.org/#/builders/46/builds/13929 "The change is no longer eligible for processing. Pull Request was already closed when EWS attempted to process it.") | [✅ 🧪 mac-wk2-stress](https://ews-build.webkit.org/#/builders/8/builds/32828 "Passed tests") | [✅ 🛠 jsc-armv7](https://ews-build.webkit.org/#/builders/35/builds/43920 "Built successfully") | 
| [✅ 🛠 🧪 merge](https://ews-build.webkit.org/#/builders/19/builds/8434 "Built successfully and passed tests") | [  ~~🛠 watch~~](https://ews-build.webkit.org/#/builders/43/builds/12865 "The change is no longer eligible for processing. Pull Request was already closed when EWS attempted to process it.") | | [❌ 🧪 jsc-armv7-tests](https://ews-build.webkit.org/#/builders/25/builds/9013 "Found 156 new JSC stress test failures: jsc-layout-tests.yaml/js/script-tests/regress-150580.js.layout, jsc-layout-tests.yaml/js/script-tests/regress-150580.js.layout-dfg-eager-no-cjit, jsc-layout-tests.yaml/js/script-tests/regress-150580.js.layout-no-cjit, jsc-layout-tests.yaml/js/script-tests/regress-150580.js.layout-no-llint, stress/array-flatmap.js.dfg-eager-no-cjit-validate, wasm.yaml/wasm/gc/any.js.default-wasm, wasm.yaml/wasm/gc/any.js.wasm-bbq, wasm.yaml/wasm/gc/any.js.wasm-collect-continuously, wasm.yaml/wasm/gc/any.js.wasm-eager, wasm.yaml/wasm/gc/any.js.wasm-eager-jettison ... (failure)") | 
| | [  ~~🛠 watch-sim~~](https://ews-build.webkit.org/#/builders/45/builds/13245 "The change is no longer eligible for processing. Pull Request was already closed when EWS attempted to process it.") | | | 
<!--EWS-Status-Bubble-End-->

mcatanzaro commented on Source/WebCore/platform/graphics/cg/UTIRegistry.cpp:
```diff
@@ -50,7 +50,6 @@ const MemoryCompactLookupOnlyRobinHoodHashSet<String>& defaultSupportedImageType
             "public.png"_s,
             "public.tiff"_s,
 #if !PLATFORM(WIN)
-            "public.jpeg-2000"_s,
```
This is enough to stop Safari from displaying JPEG 2000 images? If so, great. The Source/WebCore/platform/image-decoders/jpeg2000/ only needs to exist so long as Safari supports it.

aperezdc reviewed (commented):
I'm very glad about this, thanks!

As for the WPE and GTK ports, I believe there is no need to make the JPEG2000 support configurable at runtime&mdash;we would rather see the code removed and the OpenJPEG dependency gone 💣

mcatanzaro reviewed (dismissed):
> As for the WPE and GTK ports, I believe there is no need to make the JPEG2000 support configurable at runtime—we would rather see the code removed and the OpenJPEG dependency gone bomb

For sure, but that should not happen until _after_ the Apple code is removed. So if Apple isn't willing to remove this yet, we need to keep it too, I'm afraid. Have to remain web-compatible with Safari.

shallawa reviewed (commented):
The plan is to remove the support for JPEG2000 from all ports. For apple ports, there an API to add the support for WebView and WKWebView if needed.

shallawa committed:
Remove the support for JPEG2000
https://bugs.webkit.org/show_bug.cgi?id=178758
rdar://35161822

Reviewed by Michael Catanzaro.

Right now, no other browser is supporting JPEG2000. So there's no need for WebKit
to keep supporting it.

Clients of WebView and WKWebView can still force supporting JPEG2000 through the
private configuration/preference 'additionalSupportedImageTypes'.

* LayoutTests/fast/images/image-formats-support-expected.txt:
* LayoutTests/fast/images/image-formats-support.html:
* LayoutTests/fast/images/resources/100x100-red-jp2-renamed.png: Added.
* LayoutTests/fast/images/resources/100x100-red.jp2: Added.
* Source/WebCore/platform/ImageDecoders.cmake:
* Source/WebCore/platform/MIMETypeRegistry.cpp:
* Source/WebCore/platform/graphics/cg/UTIRegistry.cpp:
(WebCore::defaultSupportedImageTypes):
* Source/WebCore/platform/image-decoders/ScalableImageDecoder.cpp:
(WebCore::ScalableImageDecoder::create):
* Source/WebCore/platform/image-decoders/jpeg2000/JPEG2000ImageDecoder.cpp: Removed.
* Source/WebCore/platform/image-decoders/jpeg2000/JPEG2000ImageDecoder.h: Removed.
* Source/cmake/OptionsGTK.cmake:
* Source/cmake/OptionsWPE.cmake:
* Tools/TestWebKitAPI/TestWebKitAPI.xcodeproj/project.pbxproj:
* Tools/TestWebKitAPI/Tests/WebKitCocoa/100x100-red.jp2: Added.
* Tools/TestWebKitAPI/Tests/WebKitCocoa/AdditionalSupportedImageTypes.mm:
(TEST):
* Tools/TestWebKitAPI/Tests/mac/AdditionalSupportedImageTypes.html:
* Tools/TestWebKitAPI/Tests/mac/AdditionalSupportedImageTypes.mm:
(TestWebKitAPI::runTest):
(TestWebKitAPI::TEST):
* Tools/gtk/jhbuild.modules:
* Tools/wpe/jhbuild.modules:

Canonical link: https://commits.webkit.org/273978@main
